### PR TITLE
[libuuid] disallow on osx

### PR DESCRIPTION
**Pull request by autoantwort**
It conflicts with the version of the osx sdk and is not compatible and you are getting errors like [this](https://github.com/koekeishiya/homebrew-formulae/issues/9)

### CODE REVIEW

File: ports/activemq-cpp/vcpkg.json [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-  "port-version": 7,
+  "port-version": 8,
   "description": "Apache ActiveMQ is the most popular and powerful open source messaging and Integration Patterns server.",
   "supports": "!(uwp | osx)",
   "dependencies": [
     "apr",
     {
       "name": "libuuid",
-      "platform": "!windows"
+      "platform": "!windows & !osx"
```
[the remark this answers is not in the document]
Comment (LilyWangLL) [reply ports/activemq-cpp/vcpkg.json:11]: Ping @autoantwort for response

[the remark this answers is not in the document]
Comment (JackBoosY) [reply ports/activemq-cpp/vcpkg.json:11]: I think blacklisting is best here, unless upstream explicitly says that this port only supports Linux.

File: ports/drogon/mac-system-uuid.patch [hunk 2]
```diff
@@ -0,0 +1,28 @@
+diff --git a/cmake_modules/FindUUID.cmake b/cmake_modules/FindUUID.cmake
+index 31bad4c..2532829 100755
+--- a/cmake_modules/FindUUID.cmake
++++ b/cmake_modules/FindUUID.cmake
+@@ -76,18 +76,18 @@ else(UUID_LIBRARIES AND UUID_INCLUDE_DIRS)
+                      /opt/lib
+                      /usr/freeware/lib64)
+ 
+-  if(NOT UUID_LIBRARY AND BSD)
++  if(NOT UUID_LIBRARY AND (BSD OR APPLE))
```
Comment (JackBoosY) [inline ports/drogon/mac-system-uuid.patch:10]: If we drop support with APPLE, should we update this?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/drogon/mac-system-uuid.patch (hunk 2 above)
Comment (autoantwort) [reply ports/drogon/mac-system-uuid.patch:10]: This is a minimal version of https://github.com/drogonframework/drogon/pull/1152

**Review by JackBoosY — APPROVED** [review]

**Comment by strega-nil-ms** [comment]
I don't understand; it seems to work fine on my machine, and that's an error from 2019. Is this actually still an issue?

**Comment by autoantwort** [comment]
> I don't understand; it seems to work fine on my machine, and that's an error from 2019. Is this actually still an issue? Do you have any examples of things breaking more recently?

> Is this actually still an issue? Do you have any examples of things breaking more recently?

Yes this is still in issue. podofo for example fails when libuuid is installed.

**Comment by JackBoosY** [comment]
Can you please resolve the file conflicts first?

**Comment by strega-nil-ms** [comment]
Yep, alright this is correct.

**Comment by strega-nil-ms** [comment]
@autoantwort could you merge with master?

**Pull request merged**